Dockerfile

```diff
@@ -1,20 +1,53 @@
-FROM node:10-alpine
+FROM alpine:3.10 AS buildcontainer
 
-RUN apk update && apk upgrade && apk add nmap nmap-scripts
+ENV  NMAP_SHA256SUM="fcfa5a0e42099e12e4bf7a68ebe6fde05553383a682e816a7ec9256ab4773faa" \
+     NMAP_VERSION=7.80
 
-COPY package.json package-lock.json /src/
+WORKDIR /nmap
 
-WORKDIR /src
+ARG NMAP_PACKAGE="nmap-${NMAP_VERSION}.tar.bz2"
+ARG NMAP_URI="https://nmap.org/dist/${NMAP_PACKAGE}"
+
+RUN echo "Installing Nmap ${NMAP_VERSION}" && \
```

Comment:
In the build images you can be a lot more wastefull with `RUN` statements.
If you have these long chains of commands in a run statement it is often hard to debug, which command failed.

I'd recommend to group the apk installs into a single command: 

```Dockerfile
RUN apk update && \
    apk upgrade && \
    apk add build-base flex libcap-dev ...
```

The rest can all be done in separate `RUN` steps.
(Except the `cd` part, you'll have to do that by using the `WORKDIR`)

Reply:
(I'm not sure about the `set -ex` part. I don't know what it does this might also have to be chained.)